pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1464,30 +1464,9 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		}
 	}
 
-	networks := map[string]*v1.Network{}
-	cniNetworks := map[string]int{}
-	multusNetworkIndex := 1
-	for _, network := range vmi.Spec.Networks {
-		numberOfSources := 0
-		if network.Pod != nil {
-			numberOfSources++
-		}
-		if network.Multus != nil {
-			if network.Multus.Default {
-				// default network is eth0
-				cniNetworks[network.Name] = 0
-			} else {
-				cniNetworks[network.Name] = multusNetworkIndex
-				multusNetworkIndex++
-			}
-			numberOfSources++
-		}
-		if numberOfSources == 0 {
-			return fmt.Errorf("fail network %s must have a network type", network.Name)
-		} else if numberOfSources > 1 {
-			return fmt.Errorf("fail network %s must have only one network type", network.Name)
-		}
-		networks[network.Name] = network.DeepCopy()
+	networks, err := indexNetworksByName(vmi.Spec.Networks)
+	if err != nil {
+		return err
```

Comment:
The operation description/hint should come from the function/object that created it, otherwise we will have to re-do this in all callers which is a duplication.
The name of the VMI is not a fit this deep in the call stack, it is expected to be outputted to the logger, upper in the stack. I think the virt-launcher is processing everything per a VMI, so this is surely happen already.

The only missing info left now is the "index" operation. Well, I am not sure if this is really needed here. The failure in the indexing is due to the network type configuration, adding the name "index" adds little here, but if you really insist I will add it in the original error message inside `indexNetworksByName`.